core-graphics/src/data_provider.rs

```diff
@@ -50,17 +50,17 @@ impl CGDataProvider {
     ///
     /// The `CGDataProvider` object takes ownership of the reference. Once the data provider
     /// is destroyed, the reference count of the buffer is automatically decremented.
-    pub fn from_buffer(buffer: Arc<Vec<u8>>) -> Self {
+    pub fn from_buffer<T: AsRef<[u8]> + Sync + Send>(buffer: Arc<T>) -> Self {
         unsafe {
-            let ptr = (*buffer).as_ptr() as *const c_void;
-            let len = buffer.len() as size_t;
-            let info = mem::transmute::<Arc<Vec<u8>>, *mut c_void>(buffer);
-            let result = CGDataProviderCreateWithData(info, ptr, len, Some(release));
+            let ptr = (*buffer).as_ref().as_ptr() as *const c_void;
+            let len = (*buffer).as_ref().len() as size_t;
+            let info = mem::transmute::<Arc<T>, *mut c_void>(buffer);
```

Comment:
I know it's pre-existing code, but could you make this Arc::into_raw instead?

Reply:
Yeah, that's much better